source/client.c

```diff
@@ -3025,6 +3111,21 @@ static uint16_t s_aws_mqtt_client_connection_311_publish(
     return 0;
 }
 
+static uint16_t s_aws_mqtt_client_connection_311_publish(
+    void *impl,
+    const struct aws_byte_cursor *topic,
+    enum aws_mqtt_qos qos,
+    bool retain,
+    const struct aws_byte_cursor *payload,
+    aws_mqtt_op_complete_fn *on_complete,
+    void *userdata) {
+
+    struct aws_mqtt_client_connection_311_impl *connection = impl;
+
+    return aws_mqtt_client_connection_311_publish(
```

Comment:
if `qos == 0` should we not set the `timeout_ns` to `connection->operation_timeout`? That may result in timeout tasks being generated for publishes that shouldn't have them. This might be caught elsewhere but I'm not immediately seeing it.

Reply:
There's an early out in the timeout scheduling function that ignores qos 0 publishes (because packet id is 0)